src/main/java/Duke.java

```diff
@@ -1,10 +1,254 @@
+            
+        } catch (Exception e) {
+            String horizontalLine = "_______________________________________________________";
+
+            throw new MissingTaskDescriptionException(horizontalLine
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + horizontalLine);
+        }
+    }
+
+    private static Task createNewDeadline(String nextLine) throws MissingTaskDescriptionException {
+        try {
+            
+            String[] commandParts = nextLine.split("\\s", 2);
+            String[] deadlineParts = commandParts[1].split("/by");
+            LocalDate deadline = LocalDate.parse(deadlineParts[1].trim());
+            String afterDateTimeFormat = deadline.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+            Task newDeadlineTask = new Deadline(deadlineParts[0].trim(), afterDateTimeFormat);
+            return newDeadlineTask;
+            
+        } catch (Exception e) {
+            String horizontalLine = "_______________________________________________________";
+
+            throw new MissingTaskDescriptionException(horizontalLine
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + horizontalLine);
+        }
+    }
+
+    private static Task createNewTask(String nextLine) throws MissingTaskDescriptionException {
+        try {
+            
+            String[] commandParts = nextLine.split("\\s", 2);
+            String[] eventParts = commandParts[1].split("/at");
+            LocalDate event = LocalDate.parse(eventParts[1].trim());
+            String afterDateTimeFormat = event.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+            Task newEventTask = new Event(eventParts[0].trim(), afterDateTimeFormat);
+            return newEventTask;
+            
+        } catch (Exception e) {
+            String horizontalLine = "_______________________________________________________";
+
+            throw new MissingTaskDescriptionException(horizontalLine
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + horizontalLine);
+        }
+    }
+
+
+    private static int markDoneTask(String nextLine) throws MissingTaskNumberException {
+      try {
+          
+          String[] commandParts = nextLine.split("\\s", 2);
+          int taskNumber = Integer.parseInt(commandParts[1].trim()) - 1;
+          return taskNumber;
+          
+      } catch (Exception e) {
+
+          String horizontalLine = "_______________________________________________________";
+          
+          throw new MissingTaskNumberException(horizontalLine 
+                  + "\r\n"
+                  + "Oops! The task number cannot be missing :("
+                  + "\r\n"
+                  + horizontalLine);
+      }
+    } 
```

Comment:
I noticed that the indentation here is 10 spaces instead of 12 spaces. Apart from this, I believe your indentations are very well done 👍